### PR TITLE
GetSettlementType now returns enum and value is no longer configurable (hard coded in each PAM impl)

### PR DESCRIPTION
**Pull request by goodnightanderson**

### CODE REVIEW

File: pam/vplugin/vplugin.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ func RegisterGobs() {
 	gob.RegisterName("configs.TraceConfig", configs.TraceConfig{})
 
 	gob.Register(map[string]interface{}{})
+	gob.Register(new(interface{}))
```
Comment (gustavste) [inline pam/vplugin/vplugin.go:43]: not possible to register with `pam.SettlementType`? 
`gob.Register("pam.SettlementType", pam.SettlementType{})` Then that would be used in the `GetSettlementType`-function as well

Comment (goodnightanderson) [reply pam/vplugin/vplugin.go:43]: This is for the params sent to the plugin call, which in this case is practically nil as there are no input parameters to the function.... Don't think I need to register the pam.SettlementType as it is essentially just a string and not a struct, but that return type will ofc be used in the function.

**Review by gustavste — APPROVED** [review]

File: pam/vplugin/vplugin.go (hunk 1 above)
Comment (gnirb) [reply pam/vplugin/vplugin.go:43]: Just maybe add a comment why its registered, since it looks a bit strange

**Review by gnirb — APPROVED** [review]
lgtm

**Pull request merged**